Source/SwiftLintFramework/Rules/TrailingCommaRule.swift

```diff
@@ -9,14 +9,45 @@
 import Foundation
 import SourceKittenFramework
 
-private let missingTrailingCommaReason = "Multi-line collection literals should have trailing commas."
-private let extraTrailingCommaReason = "Collection literals should not have trailing commas."
+private enum TrailingCommaReason: String {
+    case missingTrailingCommaReason = "Multi-line collection literals should have trailing commas."
+    case extraTrailingCommaReason = "Collection literals should not have trailing commas."
+}
+
+private typealias CommaRuleViolation = (index: Int, reason: TrailingCommaReason)
 
-public struct TrailingCommaRule: ASTRule, ConfigurationProviderRule {
+public struct TrailingCommaRule: ASTRule, CorrectableRule, ConfigurationProviderRule {
     public var configuration = TrailingCommaConfiguration()
 
     public init() {}
 
+    private static let triggeringExamples: [String] = {
+        var result = [
+            "let foo = [1, 2, 3↓,]\n",
+            "let foo = [1, 2, 3↓, ]\n",
+            "let foo = [1, 2, 3   ↓,]\n",
+            "let foo = [1: 2, 2: 3↓, ]\n",
+            "struct Bar {\n let foo = [1: 2, 2: 3↓, ]\n}\n",
+            "let foo = [1, 2, 3↓,] + [4, 5, 6↓,]\n",
+            "let example = [ 1,\n2↓,\n // 3,\n]"
+            // "foo([1: \"\\(error)\"↓,])\n"
+            ]
+        #if !os(Linux)
+            // disabled on Linux because of https://bugs.swift.org/browse/SR-3448 and
+            // https://bugs.swift.org/browse/SR-3449
+            result.append("let foo = [\"אבג\", \"αβγ\", \"🇺🇸\"↓,]\n")
+        #endif
+        return result
+    }()
+    private static let corrections: [String: String] = {
+        let fixed = triggeringExamples.map { $0.replacingOccurrences(of: "↓,", with: "") }
```

Comment:
The correction for an excessive comma is to remove that comma. So basically yes.

Creating the correction tests algorithmically instead of writing them out individually allowed me to simply to channel it through workaround for the Linux bug, without having to write the same workaround in several places.